qt/python/mantidqt/mantidqt/widgets/sliceviewer/models/model.py

```diff
@@ -486,6 +486,21 @@ def get_proj_matrix(self):
                 for icol, idim in enumerate(i_qdims):
                     # copy first irow_end components of basis vec are always Q
                     proj_matrix[0:irow_end, icol] = list(ws.getBasisVector(idim))[0:irow_end]
+                # if determinant is zero, try to get the info from log or revert back to identity matrix
+                if np.isclose(np.linalg.det(proj_matrix),0):
+                    # for histo try to find axes from log
+                    try:
+                        expt_info = ws.getExperimentInfo(0)
+                        trans_matrix = np.array(expt_info.run().get(PROJ_MATRIX_LOG_NAME).value, dtype=float).reshape(3, 3)
+                        ndims = ws.getNumDims()
```

Comment:
I'm not sure this check is required is `W_MATRIX` is only set on 3D workspace with 3 Q-dimensions (as in `ConvertWANDSCDToQ`). If it is necessary then I believe `ndim`, `i_qdims` and `irow_end` variables are all still valid as set above L489 (and don't need to be set again).

Reply:
It will be removed